test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -0,0 +1,222 @@
+
+    public function testGloballyInstalledPluginDoesNotGenerateVersionsForLocalProject()
+    {
+        $this->createPackageVersionsArtifact();
+
+        file_put_contents($this->tempGlobalComposerHome . '/composer.json', json_encode([
+            'name'         => 'package-versions/e2e-global',
+            'require'      => [
+                'ocramius/package-versions' => '1.0.0'
+            ],
+            'repositories' => [
+                [
+                    'packagist' => false,
+                ],
+                [
+                    'type' => 'artifact',
+                    'url' => $this->tempArtifact,
+                ]
+            ]
+        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        $this->exec(__DIR__ . '/../../vendor/bin/composer global update');
+
+        $this->createArtifact();
+        file_put_contents($this->tempLocalComposerHome . '/composer.json', json_encode([
+            'name'         => 'package-versions/e2e-local',
+            'require'      => [
+                'test/package' => '2.0.0'
+            ],
+            'repositories' => [
+                [
+                    'packagist' => false,
+                ],
+                [
+                    'type' => 'artifact',
+                    'url' => $this->tempArtifact,
+                ]
+            ]
+        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        $this->execInDir('composer update', $this->tempLocalComposerHome);
+        $this->assertFileNotExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+    }
+
+    public function testRemovingPluginDoesNotAttemptToGenerateVersions()
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createArtifact();
+
+        file_put_contents($this->tempLocalComposerHome . '/composer.json', json_encode([
+           'name'         => 'package-versions/e2e-local',
+           'require'      => [
+               'test/package' => '2.0.0',
+               'ocramius/package-versions' => '1.0.0'
+           ],
+           'repositories' => [
+               [
+                   'packagist' => false,
+               ],
+               [
+                   'type' => 'artifact',
+                   'url' => $this->tempArtifact,
+               ]
+           ]
+       ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        $this->execInDir('composer update -vvv', $this->tempLocalComposerHome);
+        $this->assertFileExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+
+        $this->execInDir('composer remove ocramius/package-versions -vvv', $this->tempLocalComposerHome);
+        $this->assertFileNotExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+    }
+
+    private function createPackageVersionsArtifact()
```

Comment:
I dont think so - the path to the zip isn't used for anything, just the folder where they reside